packages/design-system/src/components/TextField/TextField.tsx

```diff
@@ -3,57 +3,31 @@ import LabelMask from './LabelMask';
 import Mask from './Mask';
 import TextInput from './TextInput';
 import classNames from 'classnames';
-import omit from 'lodash/omit';
-import pick from 'lodash/pick';
-import { FormControl, FormControlPropKeys } from '../FormControl/FormControl';
-
-// TODO: Remove `maskValue` and `unmaskValue` exports with next major release (v3.x.x)
-export { unmaskValue } from './maskHelpers';
+import { FormFieldProps, FormLabel, useFormLabel } from '../FormLabel';
 
 export type TextFieldDefaultValue = string | number;
 export type TextFieldMask = 'currency' | 'phone' | 'ssn' | 'zip';
 export type TextFieldRows = number | string;
 export type TextFieldSize = 'small' | 'medium';
 export type TextFieldValue = string | number;
-export type TextFieldErrorPlacement = 'top' | 'bottom';
 
-export interface TextFieldProps {
+export interface BaseTextFieldProps extends Omit<FormFieldProps, 'id'> {
   /**
    * Apply an `aria-label` to the text field to provide additional
    * context to assistive devices.
    */
   ariaLabel?: string;
-  /**
-   * Additional classes to be added to the root `div` element
-   */
-  className?: string;
   /**
    * Sets the initial value. Use this for an uncontrolled component; otherwise,
    * use the `value` property.
    */
   defaultValue?: TextFieldDefaultValue;
   disabled?: boolean;
-  errorMessage?: React.ReactNode;
-  /**
-   * Additional classes to be added to the error message
-   */
-  errorMessageClassName?: string;
-  /**
-   * Location of the error message relative to the field input
-   */
-  errorPlacement?: TextFieldErrorPlacement;
-  /**
-   * Additional classes to be added to the field element
-   */
   fieldClassName?: string;
-  /**
-   * Used to focus `input` on `componentDidMount()`
-   */
-  focusTrigger?: boolean;
-  /**
-   * Additional hint text to display
+  /*
+   * Sets the focus on the select during the first mount
    */
-  hint?: React.ReactNode;
+  autoFocus?: boolean;
```

Comment:
It would be good to call out this `focusTrigger` -> `autoFocus` prop change in the PR description and mark it as a breaking/major change so we don't forget to put it in our release notes